packages/profile-sync-controller/src/shared/encryption/encryption.ts

```diff
@@ -54,12 +56,14 @@ class EncryptorDecryptor {
     plaintext: string,
     password: string,
     nativeScryptCrypto?: NativeScrypt,
+    keyStore?: KeyStore,
```

Comment:
long term (far future) - I don't really like how many layers we have to drill in this context (both for NativeScrypt, and for this).

We can think of some diff approach later.

Reply:
Coming back to this, lets see if it is worth exposing an interface to persist keys.
But maybe persisting will not be necessary in this approach that uses ECIES and does not use an expensive KDF.